pkg/network/public.go

```diff
@@ -105,7 +105,7 @@ func CreatePublicNS(iface *types.PubIface, nodeID pkg.Identifier) error {
 		return err
 	}
 
-	if err := macvlan.Install(pubIface, ips, routes, pubNS); err != nil {
+	if err := macvlan.Install(pubIface, nil, ips, routes, pubNS); err != nil {
```

Comment:
now that you can pass the mac address here, I think we can change the line at https://github.com/threefoldtech/zos/blob/0fe0be7b82ffd87af3be00deb8f1d2c72e0261a8/pkg/network/public.go#L47 and instead move set mac here.